src/persist/src/client.rs

```diff
@@ -606,6 +606,30 @@ impl MultiWriteHandle {
             )));
         rx
     }
+
+    /// Atomically unblocks compaction for all contained streams up to the given `since` frontier.
+    pub fn allow_compaction_all(&self, since: Antichain<u64>) -> PFuture<SeqNo> {
+        let (tx, rx) = PFuture::new();
+
+        let client = match self.client.as_ref() {
+            Ok(client) => client,
+            Err(e) => {
+                tx.fill(Err(e.clone()));
+                return rx;
+            }
+        };
+
+        let id_sinces = self
+            .stream_ids
+            .iter()
+            .map(|id| (id.clone(), since.clone()))
```

Comment:
nit: Id should be Copy, no? I'd prefer `*id` to clone when it's applicable (eyes https://rust-lang.github.io/rust-clippy/master/#clone_on_copy)

Reply:
:+1: